pkg/api/customization/nodetemplate/formatter.go

```diff
@@ -28,30 +24,21 @@ func (ntf *Formatter) Formatter(request *types.APIContext, resource *types.RawRe
 	for _, pool := range pools {
 		if pool.Spec.NodeTemplateName == resource.ID {
 			delete(resource.Links, "remove")
-			return
+			break
 		}
 	}
-}
 
-// TODO: This should go away, it is simply a hack to get the watch on nodetemplates to filter appropriately until that system is refactored
-func filterToOwnNamespace(request *types.APIContext, resource *types.RawResource) bool {
-	user := request.Request.Header.Get("Impersonate-User")
-	if user == "" {
-		logrus.Errorf(
-			"%v",
-			httperror.NewAPIError(httperror.ServerError, "There was an error authorizing the user"))
-		return false
-	}
+	creatorID, _ := resource.Values["creatorId"].(string)
 
-	if !strings.HasPrefix(resource.ID, user+":") {
-		resource.ID = ""
-		resource.Values = map[string]interface{}{}
-		resource.Links = map[string]string{}
-		resource.Actions = map[string]string{}
-		resource.Type = ""
-
-		return false
+	user, err := ntf.UserLister.Get("", creatorID)
+	if err != nil {
+		logrus.Warnf("Failed to to get user associated with creatorId [%s]. Error: %v", creatorID, err)
```

Comment:
I'm not sure we should warn here? This would most likely indicate the user is deleted which seems fine. Perhaps an error check to validate before the warning and if it's just a 404 return out.